internal/cephfs/nodeserver.go

```diff
@@ -779,3 +786,32 @@ func (ns *NodeServer) NodeGetVolumeStats(
 
 	return nil, status.Errorf(codes.InvalidArgument, "targetpath %q is not a directory or device", targetPath)
 }
+
+// getMountOptions return the kernel/fuse mount options from CSI config file if it exists.
+// If not, it falls back to returning the kernelMountOptions/fuseMountOptions from the command line.
+func (ns *NodeServer) getMountOptions(mnt mounter.VolumeMounter, volOptions *store.VolumeOptions) (string, error) {
+	var (
+		CLIMountOptions     string
+		getMountOptionsFunc func(string, string) (string, error)
+	)
+
+	switch mnt.(type) {
+	case *mounter.FuseMounter:
+		getMountOptionsFunc = util.GetCephFSFuseMountOptions
+		CLIMountOptions = ns.fuseMountOptions
+	case *mounter.KernelMounter:
+		getMountOptionsFunc = util.GetCephFSKernelMountOptions
+		CLIMountOptions = ns.kernelMountOptions
+	}
+
+	mountOptions, err := getMountOptionsFunc(util.CsiConfigFile, volOptions.ClusterID)
+	if err != nil {
+		return "", err
+	}
+
+	if mountOptions != "" {
+		return mountOptions, nil
+	}
```

Comment:
use a single function to fetch both kernel and fuse mount options and later decide which one to pass.

```suggestion

	kernelMountOptions, fuseMountOptions, err := getMountOptions(util.CsiConfigFile, volOptions.ClusterID)
	if err != nil {
		return "", err
	}
	
	case *mounter.FuseMounter:
		mountOptions = fuseMountOptions
		CLIMountOptions = ns.fuseMountOptions
	case *mounter.KernelMounter:
		mountOptions = kernelMountOptions
		CLIMountOptions = ns.kernelMountOptions
	}

	if mountOptions != "" {
		return mountOptions, nil
	}
```